sdks/python/apache_beam/ml/inference/triton_inference_test.py

```diff
@@ -0,0 +1,242 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+import unittest
+from unittest.mock import Mock, MagicMock, patch
+
+import apache_beam as beam
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+# Protect against environments where tritonserver library is not available.
+# pylint: disable=wrong-import-order, wrong-import-position, ungrouped-imports
+try:
+  from apache_beam.ml.inference.base import PredictionResult, RunInference
+  from apache_beam.ml.inference.TritonModelHandler import TritonModelHandler
+  from apache_beam.ml.inference.TritonModelHandler import TritonModelWrapper
+except ImportError:
+  raise unittest.SkipTest('Triton dependencies are not installed')
```

Comment:
to make these test run, we can add this dep to ml_test extra. I tried to `pip install tritonserver` and it worked for me on py310, but didn't work on py311 or py312. I think we could introduce a py310_ml_test extra, similarly to the existing py312_ml_test extra, see: https://github.com/apache/beam/blob/e8b41d7664aee65fdf98b990a205b17b361ed222/sdks/python/tox.ini#L119C10-L119C15 https://github.com/apache/beam/blame/master/sdks/python/setup.py#L552C18-L552C18 to exercise these tests.